src/modules/payment/previous-payment-utils.ts

```diff
@@ -44,14 +44,16 @@ export function usePreviousPaymentMethods(): {
       if (!enabledPaymentTypes.includes(paymentMethod.paymentType))
         return false;
 
-      // Card has expired
+      const now = Date.now();
+      // Either the card registration has expired or the card itself has
+      // expired. If either of these is true, the payment method is invalid.
       const expired =
         paymentMethod.recurringPayment &&
-        parseISO(paymentMethod.recurringPayment.expiresAt).getTime() <
-          Date.now();
-      if (expired) return false;
+        (parseISO(paymentMethod.recurringPayment.expiresAt).getTime() < now ||
+          parseISO(paymentMethod.recurringPayment.cardExpiresAt).getTime() <
+            now);
 
-      return true;
+      return !expired;
```

Comment:
Nit, but I think this would be easier to read i it went something like 
```ts
if (!paymentMethod.recurringPayment) {
  return false;
}

if (parseISO(paymentMethod.recurringPayment.expiresAt).getTime() < now) {
  return false;
}

// ...

return true;
```